### ISSUE TITLE
Integrate SDK 0.9 into CI

### ISSUE THREAD

**Issue by zephyrbot** [zephyrproject-rtos/infrastructure-old#74]
**_Reported by Inaky Perez-Gonzalez:_**
Prepare the CI infrastructure to receive 0.9

- review how it was done in previous occasions
- pre-install SDK 0.9 in slaves
- introduce a switch in the CI code that selects one or another via ZEPHYR_SDK_INSTALL_DIR based on kernel commit (if so needed)

(Imported from Jira INF-73)

**Comment by zephyrbot** [comment]
**_by Genaro Saucedo Tejada:_**

installation was already requested, please clarify what does these "switch" mean:
* one is "in the CI code"
* another is "based on kernel commit"

**Comment by zephyrbot** [comment]
**_by Inaky Perez-Gonzalez:_**


Well, definitely on the CI code, wherever we export ZEPHYR_SDK_INSTALL it has to make the switch decision (0.8.2 or 0.9). 


_Anas Nashif_   do you have any preference on how the switch shall happen? 


(a) Have the code use 0.9 when building a commit older than date X?
(b) Just switch to 0.9 by date Y
(c) others?

**Comment by zephyrbot** [comment]
**_by Genaro Saucedo Tejada:_**

Please establish criteria, then reassign to me so I can implement it

**Comment by zephyrbot** [comment]
**_by Anas Nashif:_**

We need to support both SDKs in CI to be able to build older release such as the 1.6.x branch. So for master, always the latest, for a specific branch, we need to be able to use the supported SDK for that release.
In our case:
- any new code in master needs to be built with 0.9 SDK
- any code for 1.6 and 1.5, 0.8.2.


Another remark, please do not assign tasks back and forth, if you need a question answered, just ask in the comment for more details or clarification. Thanks.

**Comment by zephyrbot** [comment]
**_by Inaky Perez-Gonzalez:_**

So that's (a) then. What's the cut off for the _new code in master_?  DD/MM/YYYY HH:MM

**Comment by zephyrbot** [comment]
**_by Anas Nashif:_**

not what cut you are referring to, there is no cut, any code that goes into master needs to built with the new SDK. Anything that goes into 1.6 branch -> 0.8.2

**Comment by zephyrbot** [comment]
**_by Inaky Perez-Gonzalez:_**

Genaro, then whenever it's ready, make the switch for the master branch


The key will be to distinguish the branch -- we have to ask git to tell us who is the parent branch for the commit, because before merging it (for CI jobs), we cannot tell.


You might need to use:


```
case "$(git show-branch | cut -d\  -f1)"; in
    "[master"])
        export ZEPHYR_SDK_INSTALL_DIR=/opt/sdk-0.9;;
    "[v1.6.0-branch"])
        export ZEPHYR_SDK_INSTALL_DIR=/opt/sdk-0.8.2;;
    "*")
        BUG, abort;;
esac
```


will prolly be wrong, typed it off my head

**Comment by zephyrbot** [comment]
**_by Genaro Saucedo Tejada:_**

testing, verify and standard daily already passed.


Wating for DEBUG and ASSERT confs.

**Comment by zephyrbot** [comment]
**_by Genaro Saucedo Tejada:_**

https://jenkins.zephyrproject.org/sandbox/job/zephyr-sanity/1/console
https://jenkins.zephyrproject.org/sandbox/job/zephyr-verify2/40/console

**Comment by zephyrbot** [comment]
**_by Mark Linkmeyer:_**

_Genaro Saucedo Tejada_ , why is this being closed as "Won't Do" when it was previously in Code Review (which indicates it was implemented)?

**Comment by zephyrbot** [comment]
**_by Inaky Perez-Gonzalez:_**

The need to have both SDKs in parallel in the CI system was removed, so this was not needed anymore.
